executor/executor.cc

```diff
@@ -112,11 +113,33 @@ static void reply_handshake();
 #endif
 
 #if SYZ_EXECUTOR_USES_SHMEM
-const int kMaxOutput = 16 << 20;
+// The output region is the only thing in executor process for which consistency matters.
+// If it is corrupted ipc package will fail to parse its contents and panic.
+// But fuzzer constantly invents new ways of how to currupt the region,
+// so we map the region at a (hopefully) hard to guess address with random offset,
+// surrounded by unmapped pages.
+// The address chosen must also work on 32-bit kernels with 1GB user address space.
+const uint64 kOutputBase = 0x1b2bc20000ull;
+
+// Allocating (and forking) virtual memory for each executed process is expensive, so we only mmap
+// the amount we might possibly need for the specific received prog.
+const int kMaxOutputComparisons = 14 << 20; // < 1% of all executions.
+const int kMaxOutputCoverage = 6 << 20; // up to 1/3 of all executions (depending on corpus rotation).
+const int kMaxOutputSignal = 4 << 20;
+#ifndef SYZ_EXECUTOR_USES_FORK_SERVER
+const int kInitialOutput = kMaxOutputComparisons;
+#endif
+const int kInitialOutput = kMaxOutputSignal; // to be shared with the parent syz-executor process.
```

Comment:
If not defined SYZ_EXECUTOR_USES_FORK_SERVER - we have 2 kInitialOutput definitions. Right?

Reply:
There is something seriously broken in our testing if CI did not detect it.
The "test" OS uses all combinations of fork server/shmem:
https://github.com/google/syzkaller/blob/d0830353e30438120e98eb8b8c4c176095093fad/sys/targets/targets.go#L167-L218
And we should be building executors for all arches of "test" OS at least in some tests.
If it's indeed the case, please check why none of the tests failed.

Reply:
Thanks for noticing the problem, @tarasmadan!

In fact, this will always compile and that's why our CI did not detect it. `SYZ_EXECUTOR_USES_FORK_SERVER` is always defined and must be checked not via `#ifdef`, but via just `#if`.

https://github.com/google/syzkaller/blob/d0830353e30438120e98eb8b8c4c176095093fad/sys/syz-sysgen/sysgen.go#L377-L379

I'll fix it on this and other occasions in this patch.

Reply:
Okay, then "ifdef SYZ" looks like a pattern for:
https://github.com/google/syzkaller/blob/master/executor/style_test.go#L16
;)

Reply:
I'll create the corresponding PR :)